pkg/component/component.go

```diff
@@ -330,3 +333,67 @@ func GetDevfileInfoFromCluster(ctx context.Context, client kclient.ClientInterfa
 		Data: devfileData,
 	}, nil
 }
+
+// ListRoutesAndIngresses lists routes and ingresses created by a component
+func ListRoutesAndIngresses(client kclient.ClientInterface, componentName string, mode string) (ings []api.Host, routes []api.Host, err error) {
+	selector := odolabels.GetSelector(componentName, "app", mode, false)
+
+	ingressGVR, err := client.GetGVRFromGVK(kclient.IngressGVK)
+	if err != nil {
+		return nil, nil, fmt.Errorf("unable to determine GVR for %s: %w", kclient.IngressGVK.String(), err)
+	}
+
+	k8sIngresses, err := client.ListDynamicResources(client.GetCurrentNamespace(), ingressGVR, selector)
+	if err != nil {
+		return nil, nil, err
+	}
+	for _, u := range k8sIngresses.Items {
+		ing := &networkingv1.Ingress{}
+		err = runtime.DefaultUnstructuredConverter.FromUnstructured(u.UnstructuredContent(), ing)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		ings = append(ings, api.Host{
+			Name: ing.GetName(),
+			Hosts: func() (hosts []string) {
+				for _, rule := range ing.Spec.Rules {
+					for _, path := range rule.HTTP.Paths {
+						hosts = append(hosts, strings.Join([]string{rule.Host, path.Path}, ""))
```

Comment:
To me, since you are joining the `host` and `path` values, I (from a user standpoint) find it a bit weird that the field name is called `hosts` (in the JSON output):

```json
{
       "ingresses": [
                {
                        "name": "my-java-springboot-app",
                        "hosts": [
                                "java-springboot.example.com/",
                                "java-springboot.example.com/toto",
                                "java-springboot.example.com2/"
                        ]
                }
        ]

---Truncated---
}
```

A value like `java-springboot.example.com/toto` does not represent a `host` to me, but an absolute path.

Can we maybe change the API definition for both Ingresses and Routes in `pkg/api/component.go` to return something raw like this?

```json
{
  "ingresses": [
    {
      "name": "my-java-springboot-app",
       "rules": [
         {
           "host": "java-springboot.example.com",
           "path": "/"
         }
       ]
     }
   ]

---Truncated---
}
```

The human-readable output of `odo describe component` could remain as it is currently.

Reply:
>A value like java-springboot.example.com/toto does not represent a host to me, but looks like a complete URI (the host part is just java-springboot.example.com).

Perhaps, you're right. My initial goal was to showcase URI (what we have rn), but I mixed up the meanings of URI and host.

I feel showing the URI will be more useful than showing host and path separately, if all user wants is to simply access the application, don't you think?

Reply:
> > A value like java-springboot.example.com/toto does not represent a host to me, but looks like a complete URI (the host part is just java-springboot.example.com).
> 
> Perhaps, you're right. My initial goal was to showcase URI (what we have rn), but I mixed up the meanings of URI and host.
> 
> I feel showing the URI will be more useful than showing host and path separately, if all user wants is to simply access the application, don't you think?

On a second thought, I don't have a sure short way of determining the correct URI because at the moment, the code doesn't interpret the regex rules, it just prints them as is. I think I'll go with what you've suggested here.